include/wlr/types/wlr_input_method_v2.h

```diff
@@ -41,16 +41,34 @@ struct wlr_input_method_v2 {
 	bool client_active; // state known to the client
 	uint32_t current_serial; // received in last commit call
 
+	struct wlr_input_method_keyboard_grab_v2 *keyboard_grab;
+
 	struct wl_list link;
 
 	struct wl_listener seat_destroy;
 
 	struct {
 		struct wl_signal commit; // (struct wlr_input_method_v2*)
+		struct wl_signal grab_keyboard; // (struct wlr_input_method_v2*)
```

Comment:
The data pointer is a `struct wlr_input_method_keyboard_grab_v2 *`

Reply:
Fixed.